### PR TITLE
PWN-9424 - integrate token service and tokenExtension in all screens and features

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-9424

## Description of Work

PWN-9424 - integrate token service and tokenExtension in all screens and features

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/ui/new/NewSelectTokenPresenter.kt [hunk 1]
```diff
@@ -30,8 +32,9 @@ class NewSelectTokenPresenter(
     override fun attach(view: NewSelectTokenContract.View) {
         super.attach(view)
 
-        userTokensRepository.observeUserTokens()
-            .map(::getTokensToSelect)
+        tokenServiceCoordinator.observeUserTokens()
+            .filterIsInstance<UserTokensState.Loaded>()
+            .map { getTokensToSelect(it.solTokens) }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/home/ui/new/NewSelectTokenPresenter.kt:37]: We should be able to send `ETH` as well, don't we?

**Review by 1sakov — APPROVED** [review]
One comment

File: app/src/main/java/org/p2p/wallet/home/model/TokenConverter.kt [hunk 2]
```diff
@@ -69,18 +69,24 @@ object TokenConverter {
...
+            mintAddress = tokenMetadata.mintAddress,
+            iconUrl = tokenMetadata.iconUrl,
+            isWrapped = tokenMetadata.isWrapped,
+            rate = price?.rate?.usd,
+            tokenExtensions = TokenExtensions.NONE
         )
+        return TokenExtensionsConfigurator(
+            extensions = tokenMetadata.extensions,
+            token = token
+        ).config() as Token.Other
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/model/TokenConverter.kt:88]: just pass the <T> inside the configurator

File: app/src/main/java/org/p2p/wallet/jupiter/repository/tokens/JupiterSwapTokensRemoteRepository.kt [hunk 3]
```diff
@@ -53,18 +54,34 @@ internal class JupiterSwapTokensRemoteRepository(
...
-            chainId = response.chainId,
-            decimals = response.decimals,
-            coingeckoId = response.extensions?.coingeckoId,
-            logoUri = tokenLogoUri,
-            tokenName = response.name,
-            tokenSymbol = response.symbol,
-            tags = response.tags,
-        )
+    private suspend fun List<JupiterTokenResponse>.toJupiterToken(): List<JupiterSwapToken> = map { response ->
+        val token = userRepository.findTokenByMint(response.address)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/jupiter/repository/tokens/JupiterSwapTokensRemoteRepository.kt:58]: is this tokenMetadata?

File: app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapStateManager.kt [hunk 4]
```diff
@@ -308,11 +308,11 @@ class SwapStateManager(
     }
 
     private fun observeUserTokens() {
-        homeLocalRepository.getTokensFlow()
-            .mapLatest { userTokens ->
-                userTokensChangeHandler.handleUserTokensChange(state.value, userTokens)
+        tokenServiceCoordinator.observeUserTokens()
+            .filterIsInstance<UserTokensState.Loaded>()
+            .onEach { userTokensState ->
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapStateManager.kt:313]: try to keep mapLatest here and update new state with onEach

File: app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt [hunk 5]
```diff
@@ -205,6 +205,10 @@ class NewSendPresenter(
         }
     }
 
+    private suspend fun getNonZeroUserTokens(): List<Token.Active> {
+        return tokenServiceCoordinator.getUserTokens().filterNot { it.isZero }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt:209]: which is better?
it.isZero or it.total.isZero?

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/jupiter/repository/tokens/JupiterSwapTokensRemoteRepository.kt (hunk 3 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/jupiter/repository/tokens/JupiterSwapTokensRemoteRepository.kt:58]: no, it is Token

File: app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt (hunk 5 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt:209]: idk 🙂

File: app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapStateManager.kt (hunk 4 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapStateManager.kt:313]: okay

**Pull request merged**